api/datadoghq/v2alpha1/test/builder.go

```diff
@@ -374,6 +375,50 @@ func (builder *DatadogAgentBuilder) WithProcessDiscoveryEnabled(enabled bool) *D
 	return builder
 }
 
+// OTel Agent
+func (builder *DatadogAgentBuilder) initOtelCollector() {
+	if builder.datadogAgent.Spec.Features.OtelCollector == nil {
+		builder.datadogAgent.Spec.Features.OtelCollector = &v2alpha1.OtelCollectorFeatureConfig{}
+	}
+}
+
+func (builder *DatadogAgentBuilder) WithOTelCollectorEnabled(enabled bool) *DatadogAgentBuilder {
+	builder.initOtelCollector()
+	builder.datadogAgent.Spec.Features.OtelCollector.Enabled = apiutils.NewBoolPointer(enabled)
+	return builder
+}
+
+func (builder *DatadogAgentBuilder) WithOTelCollectorConfig() *DatadogAgentBuilder {
+	builder.datadogAgent.Spec.Features.OtelCollector.Conf = &v2alpha1.CustomConfig{}
+	builder.datadogAgent.Spec.Features.OtelCollector.Conf.ConfigData =
+		apiutils.NewStringPointer(defaultconfig.DefaultOtelCollectorConfig)
+	return builder
+}
+
+func (builder *DatadogAgentBuilder) WithOTelCollectorConfigMap() *DatadogAgentBuilder {
+	builder.datadogAgent.Spec.Features.OtelCollector.Conf = &v2alpha1.CustomConfig{}
+	builder.datadogAgent.Spec.Features.OtelCollector.Conf.ConfigMap = &v2alpha1.ConfigMapConfig{
+		Name: "user-provided-config-map",
+	}
+	return builder
+}
+
```

Comment:
Shouldn't the operator create the config map for user provided config ? Are we asking to provide configmap as input ?

Reply:
We support two ways to pass config:
- Pass configmap directly: https://github.com/DataDog/datadog-operator/blob/1a69194e7c958641288a3f0489372acb2cecfa34/api/datadoghq/v2alpha1/test/builder.go#L398 => we don't need to build configmap
- Pass config via configdata: https://github.com/DataDog/datadog-operator/blob/1a69194e7c958641288a3f0489372acb2cecfa34/api/datadoghq/v2alpha1/test/builder.go#L391 => in this path we create the config map for the user

Reply:
Example passing configmap: https://github.com/DataDog/datadog-operator/blob/1a69194e7c958641288a3f0489372acb2cecfa34/examples/datadogagent/datadog-agent-with-otel-agent-configmap.yaml

Example passing configData (we create configMap): https://github.com/DataDog/datadog-operator/blob/1a69194e7c958641288a3f0489372acb2cecfa34/examples/datadogagent/datadog-agent-with-otel-agent.yaml